src/web3/strategy/strategy.base.controller.ts

```diff
@@ -0,0 +1,97 @@
+import { Controller, Get, Logger, Param } from '@nestjs/common';
+import { StrategyBaseService } from './strategy.base.service';
+import crypto from 'crypto';
+import { StrategyMongoService } from './strategy.mongo.service';
+import { ApiParam, ApiSecurity } from '@nestjs/swagger';
+import { StrategyRequestDto } from './strategy.dtos';
+import { EvmService } from '../token-vote/evm/evm.service';
+import { SnapshotService } from '../token-vote/snapshot/snapshot.service';
+import { GraphqlService } from '../token-vote/graphql/graphql.service';
+import { formatKebab } from './strategy.utils';
+
+@ApiSecurity('api_key')
+@Controller()
+export class StrategyBaseController {
+    readonly logger = new Logger(StrategyBaseController.name);
+
+    constructor(
+        protected strategyService: StrategyBaseService,
+        protected strategyMongoService: StrategyMongoService,
+    ) {
+        // do nothing.
+    }
+
+    _getConf() {
+        return null;
+    }
+
+    _getFileName() {
+        return null;
+    }
+
+    async onApplicationBootstrap(): Promise<void> {
+        this.logger.debug('Preparing to seed database with strategies');
+
+        if (!this._getFileName() || !this._getConf()) return;
+
+        this.logger.debug(`Registering strategy file ${this._getFileName()}`);
+
+        this.logger.debug(`With configuration ${JSON.stringify(this._getConf())}`);
+
+        const hash = crypto.createHash('md5').update(this._getFileName()).digest('hex');
+
+        const filter = {
+            name: this._getConf().name,
+            endpoint: this._getConf().api_url_base + '/' + formatKebab(this._getConf().name),
+        };
+
+        await this.strategyMongoService.updateOneByIdStrategy(hash, filter);
+
+        const stale = await this.strategyMongoService.findManyStrategy({ updatedAt: { $lt: new Date(Date.now() - 10000) } });
+
+        if (stale.length > 0) {
+            this.logger.debug(`Removing stale strategies: ${stale}`);
+
+            for (const strat of stale) {
+                await this.strategyMongoService.deleteOneByIdStrategy(strat._id);
+            }
+        }
+
+    }
+
+    runStrategy(
+        params: StrategyRequestDto,
+        strategy: (accountId: string,
+                   blockHeight: number | null,
```

Comment:
not sure blockHeight can be null, unless we set a default later on based on time-equivalent

Reply:
if it's null it defaults to current block, negative number will be current block minus blockHeight, positive number will be actual block.

Reply:
ok; we'll need to ensure that all polls require a block height. They can use a negative or timestamp to calculate one, but specifically required a static height for polls.